### ISSUE TITLE
Unhandled arbitrary-null and null-arbitrary faction relation

### ISSUE THREAD

**Issue by Vectorial1024** [Vectorial1024/Desynchronized#13]
Occurs when obtaining the "faction relation" between two pawns for calculating pawn death news importance. While this is not a fatal error (the game generates a dummy relation instance), the game gives a log error for each time this happens.

Example log follows:

```
Faction  has null relation with . Returning dummy relation.
Verse.Log:Error(String, Boolean)
RimWorld.Faction:RelationWith(Faction, Boolean)
RimWorld.Faction:RelationKindWith(Faction)
Desynchronized.GeneralExtensionHelper:AlliedTo(Faction, Faction)
Desynchronized.TNDBS.Utilities.NewsImportanceExtensionHelper:GetSocialProximityScoreForOther(Pawn, Pawn)
Desynchronized.TNDBS.TaleNewsPawnDied:CalculateNewsImportanceForPawn(Pawn, TaleNewsReference)
Desynchronized.TNDBS.NewsSpreadUtility:SelectNewsRandomly(Pawn, Pawn, TaleNewsReference&)
Desynchronized.TNDBS.NewsSpreadUtility:DetermineTaleNewsToTransmit(Pawn, Pawn, SpreadMode)
Desynchronized.TNDBS.NewsSpreadUtility:SpreadNews(Pawn, Pawn, SpreadMode)
Desynchronized.Patches.PostFix_InteractionWorker:PostFix(InteractionWorker, Pawn, Pawn)
RimWorld.InteractionWorker:Interacted_Patch1(Object, Pawn, Pawn, List`1, String&, String&, LetterDef&)
RimWorld.Pawn_InteractionsTracker:TryInteractWith_Patch1(Object, Pawn, InteractionDef)
Hospitality.Harmony.TryInteractRandomly:Replacement(Pawn_InteractionsTracker, Boolean&)
RimWorld.Pawn_InteractionsTracker:TryInteractRandomly_Patch1(Object)
RimWorld.Pawn_InteractionsTracker:InteractionsTrackerTick()
Verse.Pawn:Tick_Patch1(Object)
Verse.TickList:Tick_Patch2(Object)
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()
```

**Comment by Vectorial1024** [comment]
Added additional checks to make sure the factions are non-null before calling the "relation getter", so to avoid the RimWorld error. Should be working fine.

Fix is contained in upcoming v1.5.1

**Comment by Rusturion** [comment]
https://gist.github.com/157d30dd13a6bcbd6e7be66864fea2ad

Error occurred again soon after loading.
Fairly sure this game was entirely created after 1.5.1 release.
Starting a new game now, will let you know whether the issue occurs again. May not be worth reopening this issue just yet.

**Comment by Vectorial1024** [comment]
Sigh. It seems that I have to take stronger measures to fix this bug once and for all.

Re-opening because the reported error is similar to the original error.

**Comment by Vectorial1024** [comment]
I did some tests this time, and now the problem should have been fixed.

The fix will come out in v1.5.3 .

UPDATE: New methods are written to calculate faction goodwill between two pawns. Such method should be able to handle different casess just fine.
